strategy/json/sql/postgres/single_stream_strategy.go

```diff
@@ -43,32 +45,30 @@ func (s *SingleStreamStrategy) CreateSchema(tableName string) []string {
     event_id UUID NOT NULL,
     event_name VARCHAR(100) NOT NULL,
     payload JSON NOT NULL,
-    metadata JSONB NOT NULL,
+    metadata JSONB NOT NULL, 
+    aggregate_type VARCHAR(50) NOT NULL,
+	aggregate_id UUID NOT NULL,
+	aggregate_version SMALLINT NOT NULL,
     created_at TIMESTAMP(6) NOT NULL,
     PRIMARY KEY (no),
-    CONSTRAINT aggregate_version_not_null CHECK ((metadata->>'_aggregate_version') IS NOT NULL),
-    CONSTRAINT aggregate_type_not_null CHECK ((metadata->>'_aggregate_type') IS NOT NULL),
-    CONSTRAINT aggregate_id_not_null CHECK ((metadata->>'_aggregate_id') IS NOT NULL),
     UNIQUE (event_id)
 );`,
 		tableName,
 	)
-	statements[1] = fmt.Sprintf(
-		`CREATE UNIQUE INDEX ON %s
-((metadata->>'_aggregate_type'), (metadata->>'_aggregate_id'), (metadata->>'_aggregate_version'));`,
-		tableName,
-	)
-	statements[2] = fmt.Sprintf(
-		`CREATE INDEX ON %s
-((metadata->>'_aggregate_type'), (metadata->>'_aggregate_id'), no);`,
-		tableName,
-	)
+	statements[1] = fmt.Sprintf(`CREATE UNIQUE INDEX ON %s (aggregate_id, aggregate_type, aggregate_version);`, tableName)
+	statements[2] = fmt.Sprintf(`CREATE INDEX ON %s (no, aggregate_id, aggregate_type);`, tableName)
```

Comment:
Can you please explain why you switched the column order from `aggregate_type, aggregate_id, no` to `no, aggregate_id, aggregate_type`

Reply:
As `no` has more cardinality than `aggregate_id` which has more than `aggregate_type`, so it makes sense to have it in this order for efficient index. Also I think in our business context, we mostly use `no` and id mostly to query, right?

Reply:
Actually the main amount of queries have `type` and `id` since when you lookup/load a aggregate you don't know the `no` of an event you only know the `type` and `id` this is why these where first in line.

Reply:
I see. Maybe in our use case `type` is not of interest as we've separate table for each aggregate type. We'll make it 'type', 'id' and 'no' here as we can anyway override it in our application.

Reply:
I've reverted the order of the indices